phpBB/phpbb/content_visibility.php

```diff
@@ -144,7 +144,14 @@ public function get_count($mode, $data, $forum_id)
 	*/
 	public function is_visible($mode, $forum_id, $data)
 	{
-		$is_visible = $this->auth->acl_get('m_approve', $forum_id) || $data[$mode . '_visibility'] == ITEM_APPROVED;
+		$visibility = $data[$mode . '_visibility'];
+		$poster_key = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
+		$is_visible = ($visibility == ITEM_APPROVED) ||
+			($this->config['display_unapproved_posts'] &&
+				($this->user->data['user_id'] <> ANONYMOUS) &&
+				($visibility == ITEM_UNAPPROVED || $visibility == ITEM_REAPPROVE) &&
+				($this->user->data['user_id'] === $data[$poster_key])) ||
```

Comment:
Is it guaranteed that these are the same type? At this point I would probably go with `==` to be on safe.